__tests__/test.spec.js

```diff
@@ -1,64 +1,85 @@
 import React from 'react';
-import { mount } from 'enzyme';
+import { mount, shallow } from 'enzyme';
 import renderer from 'react-test-renderer';
-import { Carousel, Container, Jumbotron } from 'react-bootstrap';
+import Carousel from 'react-bootstrap/Carousel';
+import Container from 'react-bootstrap/Container';
+import Jumbotron from 'react-bootstrap/Jumbotron';
 import App from '../client/src/app.jsx';
 import TitleImage from '../client/src/components/titleImage';
+import ProductDetails from '../client/src/components/productDetails.jsx';
 import MainImage from '../client/src/components/mainImage';
 
-// describe('test', () => {
-//   test('tests', () => {expect(true).toBe(true);})
-// })
 describe('App', () => {
-  // test('renders snapshot', () => {
-  //   const component = renderer.create(<App />);
-  //   let tree = component.toJSON();
-  //   expect(tree).toMatchSnapshot();
-  // });
+  test('renders snapshot', () => {
+    const component = renderer.create(<App />);
+    let tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+  const wrapper = mount(<App />);
   it('renders the inner Containers', () => {
-    const wrapper = mount(<App />);
-    expect(wrapper.find(Container).length).toEqual(3);
+    expect(wrapper.find(Container).length).toEqual(1);
   });
   it('renders the Title Image', () => {
-    const wrapper = mount(<App />);
     expect(wrapper.find(TitleImage).length).toEqual(1);
   });
-  it('renders the Main Image', () => {
-    const wrapper = mount(<App />);
-    expect(wrapper.find(MainImage).length).toEqual(1);
+  it('renders the Product Info module', () => {
+    expect(wrapper.find(ProductInfo).length).toEqual(1);
+  });
+  it('renders the Product Details module', () => {
+    expect(wrapper.find(ProductDetails).length).toEqual(1);
   });
 });
 
-describe('TitleImage', () => {
+describe('Title Image', () => {
   test('renders the Title Image', () => {
-    const wrapper = mount(<TitleImage />);
+    const wrapper = shallow(<TitleImage />);
     expect(wrapper.find(Jumbotron).length).toEqual(1);
   });
   it('renders the company name', () => {
-    const wrapper = mount(<TitleImage />);
+    const wrapper = shallow(<TitleImage />);
     expect(wrapper.find('h1').text()).toEqual('The Proto Company');
   });
   it('renders the informational text', () => {
-    const wrapper = mount(<TitleImage />);
+    const wrapper = shallow(<TitleImage />);
     expect(wrapper.find('p').text()).not.toBe('');
```

Comment:
is there a reason you aren't checkin that it renders the correct text? not just that it doesn't render the wrong text

Reply:
That's a good point!